app/models/metrics/base_metric.rb

```diff
@@ -0,0 +1,116 @@
+require 'csv'
+require 'json'
+
+module Metrics
+  module BaseMetric
+    attr_reader :label, :file, :interval, :timestamp,
+                :columns_name, :columns_field, :columns,
+                :rows_name, :rows_field, :rows,
+                :values
+
+    FORMATS = {
+      to_csv: 'data.csv',
+      to_fixed_key_json: 'data-fixed_key.json',
+      to_datasette_json: 'data-datasette.json',
+      to_d3_json: 'data-d3.json',
+    }.freeze
+
+    def setup_metric(metric)
+      @label = metric[:label]
+      @file = metric[:file]
+      @interval = metric[:interval]
```

Comment:
How often will the cron be triggered on production, as currently on our dev environment its set to every 30 minutes which I assume is just for testing, so I'm wondering how the timing that will affect the interval, as if its run every 30 minutes we won't need to set this?

Reply:
the cache interval per metric is there as a feature we may want to adjust and tweak in the future, to allow for some metrics to be more cached than other metrics.

At the moment the cronjob on dev is set to run every 30 mins, but I think that may shorten (and perhaps the intervals for some metrics may lengthen) once I get sense of how the data will be used and how long it takes to run.